plugin/evm/gossiper.go

```diff
@@ -521,3 +538,94 @@ func (h *GossipHandler) HandleEthTxs(nodeID ids.NodeID, msg message.EthTxsGossip
 	}
 	return nil
 }
+
+func (h *GossipHandler) HandlePullAtomicTxsGossip(nodeID ids.NodeID, msg message.PullAtomicTxsGossip) error {
+	return handlePullTxs[*GossipAtomicTx, message.AtomicTxsGossip](
+		h.vm.networkCodec,
+		h.vm.client,
+		h.atomicTxMempool,
+		nodeID,
+		msg,
+	)
+}
+
+func (h *GossipHandler) HandlePullEthTxsGossip(nodeID ids.NodeID, msg message.PullEthTxsGossip) error {
+	return handlePullTxs[*GossipEthTx, message.EthTxsGossipV2](
+		h.vm.networkCodec,
+		h.vm.client,
+		h.ethTxMempool,
+		nodeID,
+		msg,
+	)
+}
+
+func handlePullTxs[T gossip.Tx, G message.TxsGossipMessage](
+	codec codec.Manager,
+	client peer.NetworkClient,
+	mempool gossip.Mempool[T],
+	nodeID ids.NodeID,
+	msg []byte,
+) error {
+	peerFilter := &bloomfilter.Filter{}
+	if err := peerFilter.UnmarshalBinary(msg); err != nil {
+		return err
+	}
+
+	unknownTxs := mempool.GetTxs(func(tx T) bool {
+		return !peerFilter.Contains(gossip.NewHasher(tx.ID()))
+	})
+
+	txs := make([][]byte, 0, len(unknownTxs))
+	for _, tx := range unknownTxs {
+		bytes, err := tx.Marshal()
+		if err != nil {
+			return err
+		}
+		txs = append(txs, bytes)
+	}
+
+	if len(txs) == 0 {
+		return nil
+	}
+
+	nodeIDs := set.Set[ids.NodeID]{}
+	nodeIDs.Add(nodeID)
+
+	response := new(G)
+	*response = txs
+	responseBytes, err := message.BuildGossipMessage(codec, *response)
+	if err != nil {
+		return err
+	}
+
+	return client.GossipSpecific(nodeIDs, responseBytes)
+}
+
+func (h *GossipHandler) HandleAtomicTxsGossip(nodeID ids.NodeID, msg message.AtomicTxsGossip) error {
```

Comment:
hmm these are response handlers to pull requests right? If so I wonder if we should use `Request/Response` wording instead of `Gossip` in this case.

Reply:
We could do this - I avoided it initially because we aren't using AppRequest/Response and don't enforce that we send a pull gossip request before receiving the transactions. It might make more sense for us to use AppRequest/Response but we chose to use Gossip handlers because our current Request/Response clients are all blocking, and we didn't want to block on each of these.

I'm down to add Request/Response to the names though if it makes it more clear what these messages are doing.